File: backend/src/main/java/com/catijr/backend/Entities/Playlist.java
package com.catijr.backend.Entities;

import jakarta.persistence.*;
import lombok.*;

import java.time.Instant;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name="tb_playlists")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Playlist {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "playlist_id")
    private UUID id;

    @Column(name = "name")
    private String name;

    @Column(name = "description")
    private String description;

    @Column(name = "num_music")
    private int musicQtd;

    @Column(name = "duration")
    private int duration;

    @Column(name = "created_at")
    private Instant createdAt;

    @Column(name = "updated_at")
    private Instant updatedAt;

    @PrePersist
    public void onPrePersist() {
        this.createdAt = Instant.now();
    }

    @PreUpdate
    public void onPreUpdate() {
        this.updatedAt = Instant.now();
    }

    @ManyToMany
    @JoinTable(name = "tb_playlist_music", joinColumns = @JoinColumn(name = "playlist"), inverseJoinColumns = @JoinColumn(name = "songs"))
    private List<Music> songs;
}

File: frontend/src/types/playlist.ts
import type { Music } from "./music";

export interface PlaylistSummary {
  id: string;
  name: string;
  description: string | null;
  musicQtd: number;
  duration: number;
  coverImageUrls: string[];
  createdAt: string;
  updatedAt: string | null;
}

export interface Playlist extends PlaylistSummary {
  musics: Music[];
}

export interface CreatePlaylistInput {
  name: string;
  description?: string;
}

export interface UpdatePlaylistInput {
  name?: string;
  description?: string;
}